eras/shelley/impl/src/Cardano/Ledger/Shelley/Transition.hs

```diff
@@ -175,6 +246,13 @@ instance EraTransition ShelleyEra where
 
   tcInitialPParamsG = to (sgProtocolParams . stcShelleyGenesis)
 
+  toTransitionConfigKeyValuePairs stc@(ShelleyTransitionConfig _) =
+    ["shelley" .= object (toKeyValuePairs (stcShelleyGenesis stc))]
```

Comment:
Does this provide more safety than simply: 

```suggestion
  toTransitionConfigKeyValuePairs (ShelleyTransitionConfig shelleyGenesis) =
    ["shelley" .= object (toKeyValuePairs shelleyGenesis)]
```

? 
If we were to add more parameters to `ShelleyTransitionConfig` (which we probably can't anyway), we'd get a compilation error if we were using the constructor as in my suggestion, no?